moit-api/src/main/kotlin/com/mashup/moit/security/JwtTokenSupporter.kt

```diff
@@ -0,0 +1,116 @@
+package com.mashup.moit.security
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.mashup.moit.common.exception.MoitException
+import com.mashup.moit.common.exception.MoitExceptionType
+import com.mashup.moit.domain.sample.SampleUser
+import io.jsonwebtoken.JwtParser
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.security.Keys
+import org.springframework.security.oauth2.core.oidc.user.OidcUser
+import java.security.Key
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.ZoneOffset
+import java.util.Base64
+import java.util.Date
+
+class JwtTokenSupporter(key: String, mapper: ObjectMapper) {
+    private val secretKey: Key
+    private val mapper: ObjectMapper
+    private val jwtParser: JwtParser
+
+    init {
+        val encodedKey = Base64.getEncoder().encodeToString(key.toByteArray())
+        this.secretKey = Keys.hmacShaKeyFor(encodedKey.toByteArray())
+        this.mapper = mapper
+        this.jwtParser = Jwts.parserBuilder().setSigningKey(secretKey).build()
+    }
```

Comment:
```suggestion
class JwtTokenSupporter(
    key: String,
    private val mapper: ObjectMapper
) {
    private val secretKey: Key = Keys.hmacShaKeyFor(
        Base64.getEncoder().encodeToString(key.toByteArray()).toByteArray()
    )
    private val jwtParser: JwtParser = Jwts.parserBuilder().setSigningKey(secretKey).build()
```
요롷게 할 수도 있는데 어떤게 더 예쁠까요??
다른건 순서가 있으니 괜춘한데 mapper는 생성자 단에서 바로 멤버 변수로 사용해도 괜찮을것 같습니다!

Reply:
오오. mapper는 private val로 뺄 수 있겠군요. 수정하겠슴다!
다른건 코드가 너무 길어져서 init에서 초기화하는게 더 이해하는데 도움이 되지 않을까요?ㅎㅎㅎ